hack/jenkins/e2e-runner.sh

```diff
@@ -125,6 +129,18 @@ function get_latest_gci_image() {
     rm -rf .gsutil &> /dev/null
 }
 
+function get_latest_docker_release() {
+  # Typical Docker release versions are like v1.11.2-rc1, v1.11.2, and etc.
+  local -r version_re='.*\"tag_name\":[[:space:]]+\"v([0-9\.r|c-]+)\",.*'
+  local -r latest_release="$(curl -fsSL --retry 3 https://api.github.com/repos/docker/docker/releases/latest)"
+  if [[ "${latest_release}" =~ ${version_re} ]]; then
```

Comment:
I have been following the [bash 3.2 guide](http://www.tldp.org/LDP/abs/html/bashver3.html#REGEXMATCHREF) on using regular expression by not quoting them:
```
In fact, quoting in this context is not advisable as it may cause regex evaluation to fail.
```